### ISSUE TITLE
Validate recursion

### ISSUE THREAD

**Issue by samdark** [hiqdev/composer-config-plugin#48]
Currently it's possible to cause recursion with a config:

https://github.com/viktorprogger/rss-telegram-bot/blob/cc07b693bdfe31a52bf249aac840daedc50440ec/src/composer.json

That would lead to 

```
Fatal error: Allowed memory size of 1610612736 bytes exhausted (tried to allocate 20480 bytes) in D:\dev\delete_me\rss-telegram-bot\src\vendor\hiqdev\composer-config-plugin\src\Plugin.php on line 158
```

It would be good to check recursive configs in advance and error in a self-explanatory way instead.

**Comment by hiqsol** [comment]
First of all, it's senseless to include params in params cause it doesn't add anything.
But I will prevent infinite loop.

**Comment by hiqsol** [comment]
Fixed in 0ebcdea
